fsqual.cc

```diff
@@ -57,13 +79,26 @@ void test_concurrent_append(io_context_t ioctx, int fd, unsigned iodepth, std::s
     while (completed < nr) {
         auto i = unsigned(0);
         while (initiated < nr && current_depth < iodepth) {
-            io_prep_pwrite(&iocbs[i++], fd, buf, bufsize, bufsize*initiated++);
+            io_prep_pwrite(&iocbs[i], fd, buf, bufsize, bufsize*initiated++);
+            if (nowait) {
+                set_nowait_aio(iocbs[i]);
+            }
+            ++i;
             ++current_depth;
         }
         std::shuffle(iocbs.begin(), iocbs.begin() + i, random_device);
         if (i) {
             with_ctxsw_counting(ctxsw, [&] {
-                io_submit(ioctx, i, iocbps.data());
+                int left = i;
+                do {
+                    int ret = io_submit(ioctx, i, iocbps.data());
+                    if (ret == -1 && nowait && retry_nowait_aio(errno)) {
```

Comment:
This might not terminate. You're supposed to retry in a separate thread, to avoid a blocking call.

Reply:
Wait, you're not actually doing anything.

Reply:
True, this patch has to be reworked, I will close this pull request and open another when I implement this properly.